src/test/java/com/google/sps/servlets/PotentialMatchesServletTest.java

```diff
@@ -0,0 +1,233 @@
+import org.junit.runners.JUnit4;
+import org.junit.Before;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.sps.servlets.PotentialMatchesServlet;
+
+@RunWith(JUnit4.class)
+public class PotentialMatchesServletTest {
+  private static final String NO_POTENTIAL_MATCH_RESULT = "NO_POTENTIAL_MATCHES";
+
+  private static final String USER_ENTITY = "User";
+  private static final String USER_FRIENDS_LIST_PROPERTY = "friends-list";
+  private static final String USER_ID_PROPERTY = "id";
+  private static final String USER_NAME_PROPERTY = "name";
+  private static final String USER_EMAIL_PROPERTY = "email";
+  private static final String USER_BIO_PROPERTY = "bio";
+
+  private static final String TEST_USER_1_NAME = "Daenerys";
+  private static final String TEST_USER_1_ID = "5555";
+  private static final String TEST_USER_1_EMAIL = "targaryen@gmail.com";
+  private static final String TEST_USER_1_BIO = "dragons <3";
+
+  private static final String TEST_USER_2_NAME = "Eliza";
+  private static final String TEST_USER_2_ID = "1776";
+  private static final String TEST_USER_2_EMAIL = "elizaHam@gmail.com";
+  private static final String TEST_USER_2_BIO = "hamilFam";
+
+  private static final String TEST_USER_3_NAME = "Rory";
+  private static final String TEST_USER_3_ID = "1234";
+  private static final String TEST_USER_3_EMAIL = "gilmore@gmail.com";
+  private static final String TEST_USER_3_BIO = "Stars Hallow :)";
+
+  private static final String TEST_USER_4_NAME = "Steve";
+  private static final String TEST_USER_4_ID = "1964";
+  private static final String TEST_USER_4_EMAIL = "capAmerica@gmail.com";
+  private static final String TEST_USER_4_BIO = "Avengers, assemble";
+
+  private final LocalServiceTestHelper helper =
+    new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Mock
+  private HttpServletRequest mockRequest;
+ 
+  @Mock
+  private HttpServletResponse mockResponse;
+
+  private PotentialMatchesServlet servletUnderTest;
+  private DatastoreService datastore;
+ 
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    servletUnderTest = new PotentialMatchesServlet();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  /**
+  * Tests scenario where there is only one user in datastore
+  *
+  * <p>Should result in no next potential match found
+  */
+  @Test
+  public void oneUserNoPotentialMatches() throws Exception {
+    when(mockRequest.getParameter("userid")).thenReturn(TEST_USER_1_ID);
```

Comment:
`"userid"` should be a static final at the top of the file. It help prevent a typo in a new test case that costs hours of "banging your head against the wall" debugging :)

Reply:
Got it!